### ISSUE TITLE
Make segment upload to deep store misconfiguration fail fast

### ISSUE THREAD

**Issue by lrao-stripe** [apache/pinot#11211]
We ran into an issue on one of our tables to the following misconfiguration:

- Table config had upload to deep store option set
- Streaming servers did not have the segment s3 uri configuration set

This resulted in failures to upload the segments to the deep store. However, the failures only manifested during/after segment commit time. 

It'd be great to have a way to fail faster when there's a misconfiguration possibly:
-  By adding a check when the consuming segment is created and checking if the s3 uri property is configured 
-  By attempting to do this as a table level validation (this is potentially harder given that tableConfig + server config both need to be matched)

**Comment by jadami10** [comment]
for another of our clusters that we set this table config but the servers didn't have https://github.com/apache/pinot/pull/10136, it was just ignored. so that also seems like acceptable behavior

**Comment by Jackie-Jiang** [comment]
@jadami10 You mean falling back to regular controller upload? That will be more smooth (no failure), but can potentially overwhelm controller because that is the reason why we implemented the feature.
Alternatively, @navina also comes with an idea that we can write the controller deep store uri to the ZK (or open a rest api), and server can use the same uri if it is not configured

**Comment by navina** [comment]
@jadami10 Yeah. ideally, the segment store uri should never be null in the server. However, for whatever reason, we allow that to be null in the server instance. 
Falling back to controller will overwhelm controller which is what we were trying to avoid with this feature.

**Comment by jadami10** [comment]
I'm imagining you only get into this situation a few ways:
- you're turning this feature on for the first time and just got it wrong
- you have this feature on, and someone accidentally deletes the server uri or the table config

for the first one, you were already uploading via controller, so falling back seems fine. For the second one, we basically have to pick a behavior (break ingestion or potentially break the controller).

Or we need some out of band way to validate cluster configs so users can be alerted something is wrong with the setup without taking down servers/tables.

**Comment by navina** [comment]
> for the first one, you were already uploading via controller, so falling back seems fine.

You are making an assumption here that you are enabling it on existing table. For all you know, these could be new tables which can add more load to the controller.

**Comment by suddendust** [comment]
>controller deep store uri to the ZK (or open a rest api), and server can use the same uri if it is not configured

@Jackie-Jiang I am in favour of this. We shouldn't fail when we already have the information in the system.
